contracts/ChainlinkPriceOracle.sol

```diff
@@ -37,28 +58,114 @@ contract ChainlinkPriceOracle is PriceOracle, OracleErrorReporter {
         return uint(price);
     }
 
-    /*** Admin Functions ***/
+    /*** Admin Only Functions ***/
+
+    /**
+     * @notice Set a new admin for this contract
+     * @dev Only the current admin can call this function
+     * @param newAdmin The new administrator address
+     * @return Success code uint
+     */
+    function _setAdmin(address newAdmin) external returns (uint) {
+        // Check caller is admin
+        if (msg.sender != admin) {
+            return fail(Error.UNAUTHORIZED, FailureInfo.SET_ADMIN_OWNER_CHECK);
+        }
+
+        // Set new admin
+        admin = newAdmin;
+
+        emit AdminChanged(newAdmin);
+
+        return uint(Error.NO_ERROR);
+    }
+
+    /**
+     * @notice Set a new failover admin for this contract
+     * @dev Only the admin can call this function
+     * @param newFailoverAdmin The new failvover admin address
+     * @return Success code uint
+     */
+    function _setFailoverAdmin(address newFailoverAdmin) external returns (uint) {
+        // Check caller is admin
+        if (msg.sender != admin) {
+            return fail(Error.UNAUTHORIZED, FailureInfo.SET_FAILOVER_ADMIN_OWNER_CHECK);
+        }
+
+        // Set new admin
+        failoverAdmin = newFailoverAdmin;
```

Comment:
Consider checking if the address actually changed, so the event is not raised in that case.